apps/root/src/pages/transfer/components/transfer-form/index.tsx

```diff
@@ -144,6 +146,12 @@ const TransferForm = () => {
     }
   }, []);
 
+  React.useEffect(() => {
+    if (selectedNetwork.chainId !== actualCurrentNetwork.chainId && !chainIdParam) {
+      dispatch(setChainId(actualCurrentNetwork.chainId));
+    }
+  }, [actualCurrentNetwork]);
+
```

Comment:
We are having two useEffect in every app, related to setting the network. Would it be possible to merge those? I'm not sure if they will execute in the order we want every time